app/src/common/shared/org/mozilla/vrbrowser/browser/Services.kt

```diff
@@ -0,0 +1,122 @@
+import org.mozilla.vrbrowser.browser.engine.SessionStore
+import org.mozilla.vrbrowser.R
+
+class Services(context: Context, places: Places) {
+    companion object {
+        const val CLIENT_ID = "7ad9917f6c55fb77"
+        const val REDIRECT_URL = "https://accounts.firefox.com/oauth/success/$CLIENT_ID"
+    }
+
+    // This makes bookmarks storage accessible to background sync workers.
+    init {
+        RustLog.enable()
+        RustHttpConfig.setClient(lazy { HttpURLConnectionClient() })
+
+        // Make sure we get logs out of our android-components.
+        Log.addSink(AndroidLogSink())
+
+        GlobalSyncableStoreProvider.configureStore(SyncEngine.Bookmarks to places.bookmarks)
+        GlobalSyncableStoreProvider.configureStore(SyncEngine.History to places.history)
+
+        // TODO this really shouldn't be necessary, since WorkManager auto-initializes itself, unless
+        // auto-initialization is disabled in the manifest file. We don't disable the initialization,
+        // but i'm seeing crashes locally because WorkManager isn't initialized correctly...
+        // Maybe this is a race of sorts? We're trying to access it before it had a chance to auto-initialize?
+        // It's not well-documented _when_ that auto-initialization is supposed to happen.
+
+        // For now, let's just manually initialize it here, and swallow failures (it's already initialized).
+        try {
+            WorkManager.initialize(
+                    context,
+                    Configuration.Builder().setMinimumLoggingLevel(android.util.Log.INFO).build()
+            )
+        } catch (e: IllegalStateException) {}
+    }
+
+    // Process received device events, only handling received tabs for now.
+    // They'll come from other FxA devices (e.g. Firefox Desktop).
+    private val deviceEventObserver = object : DeviceEventsObserver {
+        private val logTag = "DeviceEventsObserver"
+
+        override fun onEvents(events: List<DeviceEvent>) {
+            CoroutineScope(Dispatchers.Main).launch {
+                Logger(logTag).info("Received ${events.size} device event(s)")
+                events.filterIsInstance(DeviceEvent.TabReceived::class.java).forEach {
+                    // Just load the first tab that was sent.
+                    // TODO is there a notifications API of sorts here?
+                    SessionStore.get().activeStore.loadUri(it.entries[0].url)
+                }
+            }
+        }
+    }
+
+    val accountManager = FxaAccountManager(
+        context = context,
+        serverConfig = ServerConfig.release(CLIENT_ID, REDIRECT_URL),
+        deviceConfig = DeviceConfig(
+            // This is a default name, and can be changed once user is logged in.
+            // E.g. accountManager.authenticatedAccount()?.deviceConstellation()?.setDeviceNameAsync("new name")
+            name = "${context.getString(R.string.app_name)} on ${Build.MANUFACTURER} ${Build.MODEL}",
```

Comment:
Yes, at the momen it's just going to be the app name + device and model. @jvonitter We will probably want to add some UI in the future so people can customize the device name to avoid name collision.